### PR TITLE
MLIBZ-2975: Add validation to User.signup()

### PR DESCRIPTION
**Pull request by thomasconner**

### CODE REVIEW

File: packages/js-sdk/src/user/signup.ts [hunk 1]
```diff
@@ -1,13 +1,19 @@
 import isEmpty from 'lodash/isEmpty';
+import isPlainObject from 'lodash/isPlainObject';
 import { ActiveUserError } from '../errors/activeUser';
+import { KinveyError } from '../errors/kinvey';
 import { setSession, formatKinveyBaasUrl, HttpRequestMethod, KinveyHttpRequest, KinveyBaasNamespace, KinveyHttpAuth } from '../http';
 import { getActiveUser } from './getActiveUser';
 import { User } from './user';
 
-export async function signup(data: object | User, options: { timeout?: number, state?: boolean } = {}) {
+export async function signup(data?: object | User, options: { timeout?: number, state?: boolean } = {}) {
```
Comment (heyzooi) [inline packages/js-sdk/src/user/signup.ts]: do you have a version of signup where no parameter is required?

**Comment by heyzooi** [comment]
nevermind I saw the other PR a figured out

File: packages/js-sdk/src/user/signup.ts (hunk 1 above)
Comment (heyzooi) [reply packages/js-sdk/src/user/signup.ts]: nevermind I saw the other PR a figured out

File: packages/js-sdk/src/user/signup.ts [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
   const activeUser = getActiveUser();
   const { state = true } = options;
 
+  if (data && !isPlainObject(data)) {
```
Comment (manoldonev) [inline packages/js-sdk/src/user/signup.ts]: @thomasconner Not obvious but this essentially is a breaking change introduced with a patch version of kinvey-js-sdk library and as a cascading effect it is a breaking change in kinvey-nativescript-sdk as well. See https://github.com/NativeScript/nativescript-app-templates/pull/72

Comment (thomasconner) [reply packages/js-sdk/src/user/signup.ts]: You are right that this is not that obvious and should have been a minor version bump. Thanks for pointing this out. We are working hard to make sure our releases follow the [Semantic Versioning](https://semver.org/) for our releases.

**Pull request merged**